src/middlewared/middlewared/plugins/cluster_linux/ctdb_shared_volume.py

```diff
@@ -456,25 +458,43 @@ async def create(self, job):
             self.logger.debug('Generating clustered pwenc secret')
             await self.middleware.call('clpwenc.generate_secret')
 
-        # The peers in the TSP could be using dns names while ctdb
-        # only accepts IP addresses. This means we need to resolve
-        # the hostnames of the peers in the TSP to their respective
-        # IP addresses so we can write them to the ctdb private ip file.
-        names = [i['hostname'] for i in await self.middleware.call('gluster.peer.query')]
-        ips = await self.middleware.call('cluster.utils.resolve_hostnames', names)
-        if len(names) != len(ips):
-            # this means the gluster peers hostnames resolved to the same
-            # ip address which is bad....in theory, this shouldn't occur
-            # since adding gluster peers has it's own validation and would
-            # cause it to fail way before this gets called but it's better
-            # to be safe than sorry
-            raise CallError('Duplicate gluster peer IP addresses detected.')
-
         # Setup the ctdb daemon config. Without ctdb daemon running, none of the
         # sharing services (smb/nfs) will work in an active-active setting.
-        priv_ctdb_ips = [i['address'] for i in await self.middleware.call('ctdb.private.ips.query')]
-        for ip_to_add in [i for i in ips if i not in [j for j in priv_ctdb_ips]]:
-            ip_add_job = await self.middleware.call('ctdb.private.ips.create', {'ip': ip_to_add})
+
+        priv_ctdb_ips = await self.middleware.call('ctdb.private.ips.query')
+        gluster_peers = await self.middleware.call('gluster.peer.query')
+        private_ips_nodes_uuid = set([x['node_uuid'] for x in priv_ctdb_ips])
+        gluster_peers_uuid = set([x['uuid'] for x in gluster_peers])
+
+        if not ip_info and not priv_ctdb_ips:
+            raise CallError(
+                'No private IPs were specified in payload and no private IPs are currently '
+                'set within the CTDB system directory.'
+            )
+        elif not ip_info:
+            private_ips_nodes_uuid = set([x['node_uuid'] for x in priv_ctdb_ips])
```

Comment:
Lines 475 and 476 can be removed since you're generating the same list on lines 466 and 467 respectively.